### PR TITLE
#8355: Add sweep test for logical_and_

### PR DESCRIPTION
**Pull request by mouliraj-mcw**
### Ticket
Link to Github Issue #8355 

### What's changed

- The Previousl `logical_andi` operation was designed for a tensor and a scalar. However, during the TTNN migration, it was replaced with a tensor-tensor`(logical_and_)` in-place version of the `logical_and` operation.
- As a result, I added new sweep tests for this in the TTNN infrastructure.

### Checklist
- [ ] All Post commit CI

### CODE REVIEW

File: tests/sweep_framework/sweeps/logical_and_.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+            [8, 5, 13, 512],
+            [8, 5, 32, 512],
+            [1, 1, 32, 16384],
+        ],
+        "input_a_dtype": [ttnn.bfloat16],
+        "input_b_dtype": [ttnn.bfloat16],
+        "input_a_layout": [ttnn.TILE_LAYOUT],
+        "input_b_layout": [ttnn.TILE_LAYOUT],
+        "input_a_memory_config": [ttnn.DRAM_MEMORY_CONFIG],
+        "input_b_memory_config": [ttnn.DRAM_MEMORY_CONFIG],
```
Comment (Aswinmcw) [inline tests/sweep_framework/sweeps/logical_and_.py:36]: sweep tests are meant to sweep through all possible combinations, so add bf8b, row major layout and L1 mem config

**Review by Aswinmcw — CHANGES_REQUESTED** [review]

File: tests/sweep_framework/sweeps/logical_and_.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        layout=input_b_layout,
+        device=device,
+        memory_config=input_a_memory_config,
+    )
+    start_time = start_measuring_time()
+    ttnn.logical_and_(input_tensor_a, input_tensor_b)
+    output_tensor = ttnn.to_torch(input_tensor_a)
+    e2e_perf = stop_measuring_time(start_time)
+
+    return [check_with_pcc(torch_output_tensor, output_tensor, 0.999), e2e_perf]
```
Comment (Aswinmcw) [inline tests/sweep_framework/sweeps/logical_and_.py:82]: shouldn't be `assert_equal` ?

File: tests/sweep_framework/sweeps/logical_and_.py (hunk 1 above)
Comment (mouliraj-mcw) [reply tests/sweep_framework/sweeps/logical_and_.py]: Row major is not supported. I added bfloat8_b and L1 mem config. Thankyou

**Review by Aswinmcw — APPROVED** [review]
LGTM

**Review by jdesousa-TT — APPROVED** [review]
LGTM, thanks.

**Review by sjameelTT — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Pull request merged**